client/command/loot.go

```diff
@@ -0,0 +1,500 @@
+		if err != nil {
+			fmt.Printf(Warn+"Failed to fetch loot %s\n", err)
+			return
+		}
+	} else {
+		lootType, err := lootTypeFromHumanStr(filter)
+		if err != nil {
+			fmt.Printf(Warn + "Invalid loot type see --help")
+			return
+		}
+		allLoot, err = rpc.LootAllOf(context.Background(), &clientpb.Loot{Type: lootType})
+		if err != nil {
+			fmt.Printf(Warn+"Failed to fetch loot %s\n", err)
+			return
+		}
+	}
+	displayLootTable(allLoot)
+}
+
+func lootAddLocal(ctx *grumble.Context, rpc rpcpb.SliverRPCClient) {
+	localPath := ctx.Args.String("path")
+	if localPath == "" {
+		fmt.Printf(Warn + "Missing local path argument, see --help\n")
+		return
+	}
+	if _, err := os.Stat(localPath); os.IsNotExist(err) {
+		fmt.Printf(Warn+"Path '%s' not found\n", localPath)
+		return
+	}
+
+	name := ctx.Flags.String("name")
+	if name == "" {
+		name = path.Base(localPath)
+	}
+	lootTypeStr := ctx.Flags.String("type")
+	var lootType clientpb.LootType
+	var err error
+	if lootTypeStr != "" {
+		lootType, err = lootTypeFromHumanStr(lootTypeStr)
+		if err != nil {
+			fmt.Printf(Warn+"%s\n", err)
+			return
+		}
+	} else {
+		if isTextFile(localPath) {
+			lootType = clientpb.LootType_TEXT
+		} else {
+			lootType = clientpb.LootType_BINARY
+		}
+	}
+	data, err := ioutil.ReadFile(localPath)
+	if err != nil {
+		fmt.Printf(Warn+"Failed to read file %s\n", err)
+		return
+	}
+
+	ctrl := make(chan bool)
+	go spin.Until(fmt.Sprintf("Uploading loot from %s", localPath), ctrl)
+	loot, err := rpc.LootAdd(context.Background(), &clientpb.Loot{
+		Name: name,
+		Type: lootType,
+		File: &commonpb.File{
+			Name: path.Base(localPath),
+			Data: data,
+		},
+	})
+	ctrl <- true
+	<-ctrl
+	if err != nil {
+		fmt.Printf(Warn+"%s\n", err)
+	}
+
+	fmt.Printf(Info+"Successfully added loot to server (%s)\n", loot.LootID)
+}
+
+func lootAddRemote(ctx *grumble.Context, rpc rpcpb.SliverRPCClient) {
+	session := ActiveSession.GetInteractive()
+	if session == nil {
+		return
+	}
+	remotePath := ctx.Args.String("path")
+	if remotePath == "" {
+		fmt.Printf(Warn + "Missing remote path argument, see --help\n")
+		return
+	}
+	name := ctx.Flags.String("name")
+	if name == "" {
+		name = path.Base(remotePath)
+	}
+
+	ctrl := make(chan bool)
+	go spin.Until(fmt.Sprintf("Looting remote file %s", remotePath), ctrl)
+
+	download, err := rpc.Download(context.Background(), &sliverpb.DownloadReq{
+		Request: ActiveSession.Request(ctx),
+		Path:    remotePath,
+	})
+	if err != nil {
+		ctrl <- true
+		<-ctrl
+		if err != nil {
+			fmt.Printf(Warn+"%s\n", err) // Download failed
+			return
+		}
+	}
+
+	if download.Encoder == "gzip" {
+		download.Data, err = new(encoders.Gzip).Decode(download.Data)
+		if err != nil {
+			fmt.Printf(Warn+"Decoding failed %s", err)
+			return
+		}
+	}
+
+	// Determine type based on download buffer
+	var lootType clientpb.LootType
+	if isText(download.Data) {
+		lootType = clientpb.LootType_TEXT
+	} else {
+		lootType = clientpb.LootType_BINARY
+	}
```

Comment:
We have the `--type` flag for it, but looks like we can't force a type for a "remote loot". I think we should probably use the flag to enforce it, so stuff like cleartext SSH keys can be properly tagged as `credentials` instead of `text`.

Here's an example:

```shell
sliver (PLEASED_HALF) > loot remote -T credentials /home/lesnuages/.ssh/vm

[*] Successfully added loot to server (5aec0cb3-5dd0-4702-9118-c0c14bcd0ca8)

sliver (PLEASED_HALF) > loot

Type  Name
====  ====
TEXT  vm
```

Reply:
So this is by design, but we could change it. Currently, "Credential" type loot must be structured data so they can only be added via `loot creds` or by other tools, and typically these objects don't include the `.File` field.

We could change this though, it was just to keep the code simple. Changing this just means we need to check if `.File` and `.Credentials` are `nil` more often.

Reply:
Ah gotcha, but then the `--type` flag becomes useless for this sub-command, correct?

Reply:
No, you can force TEXT or BINARY types, though I should maybe add an error if you try to force CREDENTIAL types.

Reply:
Although, now that I'm looking at the code you're right I forgot it for the `remote` subcommand lol

Reply:
On a related note, we should update the `loot` long help to include the list of valid credential types. Unless I missed it, it's not documented anywhere.

Reply:
I'm extending it to support a CREDENTIAL_FILE type right now, almost done, but will still need to add some long help.